### ISSUE TITLE
JSX tag references are reported as unused variables

### ISSUE THREAD

**Issue by fireairforce** [utooland/utoo-lint#1042]
## Version

`@utoo/lint@0.2.4`

## Minimal reproduction

```tsx
function Header() {
  return <header>Utoo</header>;
}

export function App() {
  return <Header />;
}
```

Run either rule in isolation:

```bash
npm exec --yes --package=@utoo/lint@0.2.4 -- \
  utoo-lint --rules=@typescript-eslint/no-unused-vars fixture.tsx --format=json

npm exec --yes --package=@utoo/lint@0.2.4 -- \
  utoo-lint --rules=no-unused-vars fixture.tsx --format=json
```

Both commands report the declaration on line 1:

```json
{
  "line": 1,
  "column": 10,
  "severity": "warning",
  "message": "'Header' is declared but never used."
}
```

The first command reports `@typescript-eslint/no-unused-vars`; the second reports `no-unused-vars`.

## Expected behavior

The `Header` identifier in `<Header />` is a value reference, so the declaration should be treated as used and neither rule should report it.

This also affects imported React components. In a Next.js application, imports such as `import { Header } from "./Header"` are reported as unused even when rendered as `<Header />`.

## Impact

The packaged frontend config enables both unused-variable rules. On TSX projects this produces a warning for most JSX component imports/declarations, making the preset noisy and hiding real unused-variable diagnostics. We currently have to disable both rules when adopting `@utoo/lint` in the Utoo site.

## Environment

- macOS arm64
- Node.js 22.23.1
